packages/core/theme/src/components/input.ts

```diff
@@ -58,6 +58,8 @@ const input = tv({
       "cursor-pointer",
       "active:!opacity-70",
       "rounded-full",
+      // exist end content
+      "data-[has-end-content=true]:right-9",
```

Comment:
Hi @kght6123,

Thank you for submitting the Pull Request! 🙏🏻 To ensure the `endContent` is always visible regardless of its width, I suggest we make an adjustment. We can utilize the [getBoundingClientRect](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) method on the endContent element, if it exists. By doing so, we can dynamically calculate and apply an appropriate margin-right to the clear button, factoring in the actual width of the endContent. This approach will help maintain the visibility and alignment of elements within the layout.

Reply:
@jrgarciadev 
Thank you for the great advice. I believe that to call `getBoundingClientRect`, one must reference `endContent` with React's ref and access the HTML Element. Do you have any good suggestions for this?
 I can't think of a good idea on my own, so please let me know.

Reply:
Hi @kght6123 what about adding a wrapper to the `endContent`?

```js

  const [endContentWidth, setEndContentWidth] = useState(0);

  const endContentWrapperRef = useRef(null)

  useEffect(() => {
    if (endContentRef.current) {
      setWidth(endContentRef.current.getBoundingClientRect().width);
    }
  }, [endContent]);

...
if (isClearable) {
      return (
        <div ref={endContentWrapperRef}>
          <span {...getClearButtonProps()}>{<CloseFilledIcon />}</span>
          {endContent}
        </div>
      );
    }
  ```

Reply:
thank you!

I'll try to fix it with your suggestion.

Reply:
@jrgarciadev
We have corrected the content you pointed out, so please check again.